### PR TITLE
Generalize texture format selection for SCREEN_TEXTURE

### PR DESCRIPTION
**Pull request by kklobe**
When using `output = texture`, we can query the SDL_RenderDriver's preferred texture format, rather than hardcoding.

For example, the "opengles" driver already has the only allowed format in its texture_formats[] array (from SDL_render_gles.c in the SDL source)

```
SDL_RenderDriver GLES_RenderDriver = {
    GLES_CreateRenderer,
    {
     "opengles",
     (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC),
     1,
     {SDL_PIXELFORMAT_ABGR8888},
     0,
     0
    }
};
```

Here is SDL_render_gles2.c:

```
SDL_RenderDriver GLES2_RenderDriver = {
    GLES2_CreateRenderer,
    {
        "opengles2",
        (SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC | SDL_RENDERER_TARGETTEXTURE),
        4,
        {
        SDL_PIXELFORMAT_ARGB8888,
        SDL_PIXELFORMAT_ABGR8888,
        SDL_PIXELFORMAT_RGB888,
        SDL_PIXELFORMAT_BGR888
        },
        0,
        0
    }
};

```

### CODE REVIEW

**Comment by kcgen** [comment]
Thanks for pulling those references showing no impact to the OpenGLES2 backend. 
I also like that this off-loads more decisions into SDL, which is the more informed side versus DOSBox.
Regression testing underway :-)

**Comment by kcgen** [comment]
Linux texture_renderer tests:
- [x] software
- [x] opengl
- [x] opengles2

Raspberry Pi
- [x] software
- [x] opengles2

Windows texture_renderer tests:
- [ ] software
- [ ] opengl
- [ ] direct3d

**Comment by kcgen** [comment]
@kklobe , can you share the backends you were able to test on Apple hardware?

**Comment by kklobe** [comment]
> @kklobe , can you share the backends you were able to test on Apple hardware?

Added inline above.

File: src/gui/sdlmain.cpp [hunk 1]
```diff
@@ -1074,26 +1074,21 @@ Bitu GFX_SetSize(Bitu width, Bitu height, Bitu flags,
 			LOG_MSG("SDL:Can't create renderer, falling back to surface");
 			goto dosurface;
 		}
-		/* SDL_PIXELFORMAT_ARGB8888 is possible with most
-		rendering drivers, "opengles" being a notable exception */
-		sdl.texture.texture = SDL_CreateTexture(sdl.renderer, SDL_PIXELFORMAT_ARGB8888,
+		/* Use renderer's default format */
+		SDL_RendererInfo rinfo;
+		SDL_GetRendererInfo(sdl.renderer, &rinfo);
+		const Uint32 textureFormat = rinfo.texture_formats[0];
```
Comment (kcgen) [inline src/gui/sdlmain.cpp:1080]: Suggest `const uint32_t texture_format = rinfo.texture_formats[0];`

Lowercase_snake format for local variables. 

Unless the API's types are truly unique, we also prefer to just use the standard fixed-types - it's more consistent to read. Alternative is to use `auto` as well.

Comment (kcgen) [reply src/gui/sdlmain.cpp:1080]: Sorry, I don't think I sent this your way yet :) https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md#coding-style

**Review by kcgen — COMMENTED** [review]
Looks great @kklobe, thank you!
Will merge shortly when this one touch-up is in.

File: src/gui/sdlmain.cpp (hunk 1 above)
Comment (kklobe) [reply src/gui/sdlmain.cpp]: oh, good, I like `auto`, done.

Comment (kklobe) [reply src/gui/sdlmain.cpp]: I'm not seeing snake case further down, though - should I change those as well? e.g.

```
Uint32 pixelFormat;
		SDL_QueryTexture(sdl.texture.texture, &pixelFormat, NULL, NULL, NULL);
```
```
				// does program need to be rebuilt?
				if (sdl.opengl.program_object == 0) {
					GLuint vertexShader, fragmentShader;
```

Comment (kcgen) [reply src/gui/sdlmain.cpp]: Yes, inheritted code from upstream doesn't follow a consistent convention.. each of their devs has or had a slightly different style. 

But given this is the same variable for the same purpose, then yes - if you'd like to make it consistent in the couple other spots, that would be a nice readability improvement!

Comment (kklobe) [reply src/gui/sdlmain.cpp]: > Sorry, I don't think I sent this your way yet :) https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md#coding-style

Hint taken, thank you!

Comment (kcgen) [reply src/gui/sdlmain.cpp]: After this, can you squashed these two fixes with the original commit? Then we'll merge.

**Pull request merged**